packages/react-router/lib/types/route-module.ts

```diff
@@ -109,11 +114,31 @@ type _CreateActionData<ServerActionData, ClientActionData> = Awaited<
 >
 
 type ClientDataFunctionArgs<T extends RouteInfo> = {
+  /** A {@link https://developer.mozilla.org/en-US/docs/Web/API/Request Fetch Request instance} which you can use to read headers (like cookies, and {@link https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams URLSearchParams} from the request. */
```

Comment:
Since this is a `request` created on the client it won't have the same headers/cookie access the server `request` has.  We should either call that out or just don't specifically mention those and instead refer to the commonly used client-side aspects like `request.url`

Reply:
Hmm... This type is used as the basis for `ServerDataFunctionArgs`. Does it make sense to define `request` separately on both of them so the docs can be specific to each?

Also, do you know what properties would be available on `request` for client data functions? I can't seem to find the place where the request object is generated for client data functions.

Reply:
Excellent. I'll get that sorted out soon. Thanks for all the info.